rca/people/models.py

```diff
@@ -872,6 +880,44 @@ def student_related_links(self):
             for item in self.relatedlinks.all()
         ]
 
+    def save(self, *args, **kwargs):
+        """On saving the student page, make sure the student_user_account
+        has a group created with the necessary permissions
+        """
+        super(StudentPage, self).save()
+        if self.student_user_image_collection and self.student_user_image_collection:
+            # Create a specific group for this student so they have edit access to their page
+            # and their image collection
+            specific_student_group, created = Group.objects.get_or_create(
+                name=f"Student: {self.first_name} {self.last_name}"
+            )
+
+            if created:
+                # Create new add GroupPagePermission
+                GroupPagePermission.objects.create(
+                    group=specific_student_group, page=self, permission_type="edit"
+                )
+
+                # Create new GroupCollectionPermission for Profile Images collection
+                GroupCollectionPermission.objects.create(
+                    group=specific_student_group,
+                    collection=Collection.objects.get(
+                        name=self.student_user_image_collection
+                    ),
+                    permission=Permission.objects.get(codename="add_image"),
+                )
+                GroupCollectionPermission.objects.create(
+                    group=specific_student_group,
+                    collection=Collection.objects.get(
+                        name=self.student_user_image_collection
+                    ),
+                    permission=Permission.objects.get(codename="choose_image"),
+                )
+
+            # Add the new specific student group to the user
+            self.student_user_account.groups.add(specific_student_group)
+            self.student_user_account.save()
+
```

Comment:
This needs covering by a unit test

Reply:
This isn't going to happen as part of this task. Due to all the custom form things and so many moving parts I'm not able to even test form.valid() at the moment.

Reply:
TODO - finish tests for creating page with groups